examples/tensorflow_hello_world/helloworld.py

```diff
@@ -1,11 +1,211 @@
+from xain.fl.coordinator.aggregate import FederatedAveragingAgg
+from xain.fl.participant import ModelProvider, Participant
+
+# TODO: this feels a bit hacky. it's the only flag that needs to be added here.
+#  any better solution?
+FLAGS = flags.FLAGS
+flags.DEFINE_string("task_name", None, "")
+
+
+# We first define some constants for this specific example
+
+# Specify a dataset name for this example
+# This comes from the Fashion MNIST dataset:
+# https://research.zalando.com/welcome/mission/research-projects/fashion-mnist/
+# 100p means that it is split into 100 partitions, which are IID
+# Each partition represents the dataset a single client stores locally
+DATASET_NAME = "fashion-mnist-100p-iid-balanced"
+
+# standard attributes of the Fashion MNIST dataset
+NUM_CLASSES = 10
+INPUT_SHAPE = (28, 28, 1)
+
+# R is the number of global epochs the model is going to be trained for
+R = 2
+
+# E is the number of local epochs
+# Each Participant in a round will train the model with its local data for E epochs
+E = 2
+
+# C is the fraction of total clients that participate in a training round
+C = 0.02
+
+# B is the local batch size for a client update
+B = 10
+
+# default values for exponential decay learning rate
+DEFAULT_LR = 0.002
+DEFAULT_K = 0.01
+
+
+def main(_):
+    # fetch the data
+    # xy_train_partitions: each represent one local training data on a client
+    # xy_validation contains the global validation data (used for each participant)
+    # xy_test contains the global test data
+    xy_train_partitions, xy_validation, xy_test = load_splits(DATASET_NAME)
+
+    # declare model and compile Keras model
+    model_fn = create_and_compile_model
+
+    # pass model and learning rate functions to ModelProvider
+    model_provider = ModelProvider(model_fn=model_fn, lr_fn_fn=learning_rate_fn)
+
+    # init the Participant for each client.
+    # At this stage they are not yet selected for training
+    potential_participants = init_participants(
+        xy_train_partitions=xy_train_partitions,
+        model_provider=model_provider,
+        xy_validation=xy_validation,
+    )
+
+    # init the centralized Coordinator of the training
+    coordinator = init_coordinator(
+        participants=potential_participants,
+        xy_train_partitions=xy_train_partitions,
+        model_provider=model_provider,
+        xy_validation=xy_validation,
+    )
+
+    # printing some basic statistics
+    print_dataset_stats(xy_train_partitions, xy_validation, xy_test)
+
+    # calling fit on the coordinator starts the training
+    # and print validation metrics after each global round
+    validation_metrics, _, _, _ = coordinator.fit(num_rounds=R)
+    print_validation_metrics(validation_metrics)
+
+    # evaluate the model with the test set
+    loss, accuracy = coordinator.evaluate(xy_test)
+    print(f"---\nTest completed!\nLoss: {loss} | Test accuracy: {accuracy}")
+
+
+# specify learning rate function, in this case with exponential decay
```

Comment:
We can move the comments on functions inside the function (using Pythons function-level docstring)